a10_octavia/controller/worker/controller_worker.py

```diff
@@ -1502,3 +1502,45 @@ def _get_flavor_data(self, flavor_id):
             flavor_data = json.loads(flavor_profile.flavor_data)
             return flavor_data
         return None
+
+    def delete_load_balancer_with_housekeeping(self, pending_lb, cascade=True):
+        """Function to delete load balancer for A10 provider using Housekeeper thread"""
+        if pending_lb.project_id in CONF.hardware_thunder.devices:
+            try:
+                self._lb_repo.update(db_apis.get_session(),
+                                     pending_lb.id,
+                                     provisioning_status=constants.ERROR)
+                vthunder = self._vthunder_repo.get_vthunder_from_lb(
+                    db_apis.get_session(), pending_lb.id)
+                if vthunder is not None:
+                    self._vthunder_repo.update(
+                        db_apis.get_session(),
+                        vthunder.id,
+                        status=constants.ACTIVE)
+            except Exception as e:
+                LOG.exception("Failed to update load balancer %(lb) "
+                              "provisioning status to ERROR due to: "
+                              "%(except)s", {'lb': pending_lb.id, 'except': e})
+                raise e
+            lb = self._lb_repo.get(db_apis.get_session(), id=pending_lb.id)
```

Comment:
Please help me understand why we get lb(pending_lb) and vthunder again instead of reuse them?

Reply:
We are getting the lb and vthunder again because we want an updated object of them whose status is updated as ERROR and ACTIVE respt.
If we use those objects as it is we won't get their updated status.